plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -3273,7 +3274,7 @@ public boolean addNetworkRules(final String vmName, final String vmId, final Str
         }
 
         final String newRules = rules.replace(" ", ";");
-        final Script cmd = new Script(_securityGroupPath, _timeout, s_logger);
+        final Script cmd = new Script(_securityGroupPath, _timeout.getMillis(), s_logger);
```

Comment:
Please consider adding an overridden version of the `Script(String, int, Logger)` constructor that accepts `Duration` to encapsulate this type conversion.